src/components/TodoApp.tsx

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useEffect } from 'react';
+import classNames from 'classnames';
+import { Header } from './Header';
+import { Main } from './TodoList';
+import { Footer } from './Footer';
+import { TodoContext } from '../contexts/TodoContext';
+
+export const TodoApp: React.FC = () => {
+  const { todos, error, setError } = useContext(TodoContext);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setError('');
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [error, setError]);
```

Comment:
create a component `ErrorNotification` for error jsx and put this logic with setTimeout inside  `ErrorNotification`  component